tests/roles/manila_adoption/tasks/main.yaml

```diff
@@ -1,55 +1,17 @@
-- name: deploy podified Manila with cephfs backend
-  when: manila_backend == "cephfs"
-  ansible.builtin.shell: |
-    {{ shell_header }}
-    {{ oc_header }}
-    oc patch openstackcontrolplane openstack --type=merge --patch '
-    spec:
-      manila:
-        enabled: true
-        apiOverride:
-          route: {}
-        template:
-          databaseInstance: openstack
-          databaseAccount: manila
-          manilaAPI:
-            customServiceConfig: |
-              [DEFAULT]
-              enabled_share_protocols=cephfs
-            replicas: 1
-            networkAttachments:
-            - internalapi
-            override:
-              service:
-                internal:
-                  metadata:
-                    annotations:
-                      metallb.universe.tf/address-pool: internalapi
-                      metallb.universe.tf/allow-shared-ip: internalapi
-                      metallb.universe.tf/loadBalancerIPs: 172.17.0.80
-                  spec:
-                    type: LoadBalancer
-          manilaScheduler:
-            replicas: 1
-          manilaShares:
-            share1:
-              customServiceConfig: |
-                [DEFAULT]
-                enabled_share_backends=cephfs
-                enabled_share_protocols=cephfs
-                [cephfs]
-                driver_handles_share_servers=False
-                share_backend_name=cephfs
-                share_driver=manila.share.drivers.cephfs.driver.CephFSDriver
-                cephfs_conf_path=/etc/ceph/ceph.conf
-                cephfs_auth_id=openstack
-                cephfs_cluster_name=ceph
-                cephfs_volume_mode=0755
-                cephfs_protocol_helper_type=CEPHFS
-              replicas: 1
-              networkAttachments:
-              - storage
-    '
+- name: Deploy Podified Manila
+  when: manila_backend == "cephfs" or manila_backend == "cephnfs"
+  block:
+    - name: generate CR config based on the selected backend
+      ansible.builtin.template:
+        src: manila_cephfs.yaml.j2
+        dest: /tmp/manila_cephfs.yaml
```

Comment:
just wondering if it would be useful to have that somewhere other than /tmp so we can get it in logs for debug

Reply:
I was discussing a similar thing earlier in the patch with @jistr, but in general this template is rendered and applied, which means I will find the result of what is applied in the main manila CR collected by must-gather. However I think it would be useful, as a future improvement, to put this resulting template elsewhere (maybe a path we collect in CI logs).
Let me know if you have thoughts or we did something similar for other components and I'll follow the pattern.
I might follow the same pattern I saw here [1] and render templates under `../config/`.

[1] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/tests/roles/backend_services/tasks/main.yaml#L64